### PR TITLE
engine: try cleanup transports

### PR DESCRIPTION
**Pull request by ltalirz**
@sphuber : 

~~this gets rid of a leftover transport reference~~
No it doesn't - I had a "bug" in my tests ;-)

### CODE REVIEW

File: aiida/engine/transports.py [hunk 1]
```diff
@@ -23,11 +23,48 @@
...
+            # The user still wants the transport so open it
+            _LOGGER.debug('Transport request opening transport for %s', self._authinfo)
+            try:
+                self._transport.open()
+            except Exception as exception:  # pylint: disable=broad-except
+                _LOGGER.error('exception occurred while trying to open transport:\n %s', exception)
+                self.future.set_exception(exception)
+
+                # Cleanup of the stale TransportRequest with the excepted transport future
+                # self._transport_requests.pop(authinfo.id, None)
```
Comment (ltalirz) [inline aiida/engine/transports.py:51]: this is what would have happened in the previous codebase.

does this cleanup really need to happen here or is it ok to do it at the end?
I guess all the other context managers waiting for this transport will anyhow get the excepted one.

on the other hand it's true - if this is cleaned up here already, follow-up transport requests can already get a fresh one

it just makes the code more difficult to reason about

**Comment by chrisjsewell** [comment]
Thanks @ltalirz, is there a unit test(s) we can add to check that there are indeed to leftover transport requests (one that would fail with the current code)?

**Comment by chrisjsewell** [comment]
While we are at it, it might also be good to add a brief entry somewhere in docs/source/internals, with an explanation of how the transport works, and/or in the module docstring of aiida/engine/transports.py

**Comment by ltalirz** [comment]
Yes, I'll try to add some tests later... so far this is really just a basis for discussion

Since we're seeming to get closer to the root issue, I'll try to look at that first

**Comment by ltalirz** [comment]
And indeed i had a bug in my checks - the transport and transport request  is still there.
Let's leave this cleanup Pr for later

**Pull request closed without merging**